tests/no-useless-iife.test.js

```diff
@@ -0,0 +1,57 @@
+const { RuleTester } = require('eslint')
+const rule = require('../rules/no-useless-iife')
+
+new RuleTester({
+  parserOptions: { ecmaVersion: 2021 },
+}).run('no-useless-iife', rule, {
+  valid: [
+    'const noop = () => {}',
+    'const noop = (() => {})',
+    'new Promise(() => {})',
+    'someFunction(() => {})',
+    `(() => {
+        oneInstruction
+        anotherInstruction
+      })()`,
+    `(() => {
+        if (condition) 
+          someInstruction
+      })()`,
+    `(() => {
+        try {
+          someInstruction
+        } catch {}
+      })()`,
+    `(() => {
+        while(true);
+      })()`,
+    `(() => {
+        for(;;) { someInstruction }
+      })()`,
+  ],
+  invalid: [
+    {
+      code  : '(async () => { await promise })()',
```

Comment:
Do we have top-level `await` support yet in our Node.js versions? If not, this can be useful for starting async scripts

Reply:
TLA is supported in all Node.js versions that are not EOL. Note that this rule does not forbid the use of IIFEs, it forbids the use of IIFEs that have only 0 or 1 instructions (because there's literally no good reason to use them to wrap only one instruction, regardless of TLA btw, it's just memory overhead AFAICT).